flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/result/ExecutionResult.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.gateway.service.result;
+
+import org.apache.flink.table.gateway.common.results.ResultSet;
+
+import static org.apache.flink.table.gateway.service.utils.Constants.FETCH_ALL;
+
+/** Describe the execution results. */
+public class ExecutionResult {
+
+    private final ResultFetcher resultFetcher;
+
+    public ExecutionResult(ResultFetcher fetcher) {
+        this.resultFetcher = fetcher;
+    }
+
+    public ResultSet fetchResults(long token, int maxRows) {
+        if (maxRows != FETCH_ALL && maxRows <= 0) {
+            throw new IllegalArgumentException(
```

Comment:
```suggestion
            throw new SqlGatewayException(
```

Reply:
I think we'd better throw clear exception here and let outer to wrap the exception to SqlGatewayException.